components/collections/src/codepointinvliststringlist/mod.rs

```diff
@@ -0,0 +1,310 @@
+            if let Some(mut x) = it.next() {
+                if x.len() == 1 {
+                    return Err(CodePointInversionListStringListError::InvalidStringLength(
+                        x.to_string(),
+                    ));
+                }
+                for y in it {
+                    if x.len() == 1 {
+                        return Err(CodePointInversionListStringListError::InvalidStringLength(
+                            x.to_string(),
+                        ));
+                    } else if x == y {
+                        return Err(CodePointInversionListStringListError::StringListNotUnique(
+                            x.to_string(),
+                        ));
+                    } else if x > y {
+                        return Err(CodePointInversionListStringListError::StringListNotSorted(
+                            x.to_string(),
+                            y.to_string(),
+                        ));
+                    }
+
+                    // Next window begins. Update `x` here, `y` will be updated in next loop iteration.
+                    x = y;
+                }
+            }
+        }
+
+        Ok(CodePointInversionListStringList {
+            cp_inv_list,
+            str_list,
+        })
+    }
+
+    #[doc(hidden)]
+    pub const fn from_parts_unchecked(
+        cp_inv_list: CodePointInversionList<'data>,
+        str_list: VarZeroVec<'data, str>,
+    ) -> Self {
+        CodePointInversionListStringList {
+            cp_inv_list,
+            str_list,
+        }
+    }
+
+    /// Returns the number of elements in this set (its cardinality).
+    /// Note than the elements of a set may include both individual
+    /// codepoints and strings.
+    pub fn size(&self) -> usize {
+        self.cp_inv_list.size() + self.str_list.len()
+    }
+
+    /// Return true if this set contains multi-code point strings or the empty string.
+    pub fn has_strings(&self) -> bool {
+        !self.str_list.is_empty()
+    }
+
+    ///
+    /// # Examples
+    /// ```
+    /// use icu_collections::codepointinvlist::CodePointInversionList;
+    /// use icu_collections::codepointinvliststringlist::CodePointInversionListStringList;
+    /// use zerovec::VarZeroVec;
+    ///
+    /// let cp_slice = &[0, 0x0080, 0xFFFF, 0x1_0000, 0x10_FFFF, 0x11_0000];
+    /// let cp_list =
+    ///    CodePointInversionList::try_clone_from_inversion_list_slice(cp_slice).unwrap();
+    /// let str_slice = &["", "ascii_max", "bmp_max", "unicode_max", "zero"];
+    /// let str_list = VarZeroVec::<str>::from(str_slice);
+    ///
+    /// let cpilsl = CodePointInversionListStringList::try_from(cp_list, str_list).unwrap();
+    ///
+    /// assert!(cpilsl.contains("ascii_max"));
+    /// assert!(cpilsl.contains(""));
+    /// assert!(cpilsl.contains("A"));
+    /// assert!(!cpilsl.contains("bazinga!"));
+    /// ```
+    pub fn contains(&self, s: &str) -> bool {
+        if s.len() == 1 {
```

Comment:
Issue: `str::len` returns byte length, not char length

https://doc.rust-lang.org/std/primitive.str.html#method.len

I suggest getting the first item out of `s.chars()`; if there is no second item, check the inversion list; but if there is a second item or if the string is empty, check the varzerovec.

Please add some non-ASCII tests to exercise this

Reply:
Done.